src/main/java/bio/overture/ego/config/DefaultProviderConfig.java

```diff
@@ -0,0 +1,48 @@
+package bio.overture.ego.config;
+
+import static bio.overture.ego.utils.Strings.isDefined;
+import static com.google.common.base.Preconditions.checkState;
+
+import bio.overture.ego.model.enums.ProviderType;
+import bio.overture.ego.service.DefaultProviderService;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.ApplicationListener;
+import org.springframework.context.event.ContextRefreshedEvent;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Component
+public class DefaultProviderConfig implements ApplicationListener<ContextRefreshedEvent> {
```

Comment:
This name is missleading. I would suggest something like `DefaultProviderGuardListener`